### PR TITLE
Additional Point Source Capabilities and More

### PR DESCRIPTION
**Pull request by smericks**
- Can use apparent or absolute magnitude in light components (don't think its backwards compatible, will need review/discussion)
- add csv_path option for train_model.py
- new data generation options(option to store caustic area, option to select for quads only, option to select for high magnification of point source images)
- allow for custom pixel grid kwargs
- new sampling distributions for lens center

### CODE REVIEW

**Comment by smericks** [comment]
- kwargs_synthesis=None needs to be set when calling lenstronomy.LensModel.profile_list_base_import_class() (not backwards compatible w/ older lenstronomy versions)
- numpy typing (i.e. np.int, np.bool) is no longer supported

File: paltas/Sampling/distributions.py [hunk 1]
```diff
@@ -408,3 +495,111 @@ def __call__(self):
...
+	"""
+
+	def __init__(self,psf_filepath,coords_dict,psf_option='emp_f814w'):
+		self.psf_filepath = psf_filepath
+		with fits.open(self.psf_filepath) as hdu:
+			self.psf_kernels = hdu[0].data
+		self.coords_dict = coords_dict
+		self.psf_option = psf_option
+
+	def hst_emp_f814w_mapping(self,coords):
```
Comment (JelleAalbers) [inline paltas/Sampling/distributions.py:526]: This code looks a little scary: lots of math, hardcoded numbers, print statements, etc. It also seems pretty hubble specific. If there is no way to use an external library for some of this (e.g. maybe numpy or scipy for the interpolation?), you could move the bulk of this function to a separate function in `hubble_utils.py` that you can call here. And Sebastian would want you to add unit tests for this :-)

File: paltas/Utils/cosmology_utils.py [hunk 2]
```diff
@@ -86,15 +86,15 @@ def absolute_to_apparent(mag_absolute,z_light,cosmo,
 	"""Converts from absolute magnitude to apparent magnitude.
 
 	Args:
-		mag_apparent (float): The absolute magnitude
+		mag_absolute (float): The absolute magnitude
```
Comment (JelleAalbers) [inline paltas/Utils/cosmology_utils.py:89]: Thanks for fixing this very misleading docstring! 😅

File: paltas/Utils/cosmology_utils.py [hunk 3]
```diff
@@ -109,6 +109,34 @@ def absolute_to_apparent(mag_absolute,z_light,cosmo,
 
 	return mag_apparent
 
+def apparent_to_absolute(mag_apparent,z_light,cosmo,include_k_correction=True):
```
Comment (JelleAalbers) [inline paltas/Utils/cosmology_utils.py:112]: You could avoid some duplication here by making one function (maybe starting with an underscore) that outputs the apparent-absolute offset and then call it from both apparent_to_absolute and absolute_to_apparent. That would make it immediately obvious there is not e.g. a minus sign error somewhere.

File: test/point_source_tests.py [hunk 4]
```diff
@@ -10,19 +10,21 @@
...
+		self.c = PointSourceBase(cosmology_parameters='planck18',
+			point_source_parameters=dict())
 
 	def test_update_parameters(self):
 		# test passing None
 		self.c.update_parameters(None)
 		self.assertDictEqual(self.c.point_source_parameters, dict())
 		# test passing an element that wasn't there yet
-		self.c.update_parameters({'radius':1.})
+		self.c.update_parameters(None,{'radius':1.})
```
Comment (JelleAalbers) [inline test/point_source_tests.py:27]: Maybe add a comment here about what the None is doing?

File: test/analysis_tests.py [hunk 5]
```diff
@@ -1780,7 +1780,8 @@ def test_main(self):
 		Analysis.train_model.main()
 
 		# Cleanup the files we don't want
-		os.remove('test_data/fake_model.h5')
+		os.remove('test_data/fake_model_best.h5')
```
Comment (JelleAalbers) [inline test/analysis_tests.py:1783]: I may very well have missed it, but where do these new files get made?

**Review by JelleAalbers — COMMENTED** [review]
Thanks for this Sydney, hope we can get your improvements in soon! I haven't gone through the Analysis and Sersic changes yet, but maybe these comments are already useful.

(We discussed offline that you would try to make the Sersic changes backwards compatible, e.g. by adding an optional argument that says whether the magnitude is specified as absolute or apparent.)

**Comment by smericks** [comment]
Closing this large PR to make separate PRs for each feature.

**Pull request closed without merging**